File: 1781. Sum of Beauty of All Substrings/Solution.cpp
#include <algorithm>
#include <string>
using namespace std;

class Solution {
public:
    int beautySum(string s)
    {
        int n = s.size(), res = 0;
        for (int i = 0; i < n; i++) {
            int cnt[26] = { 0 };
            for (int j = i; j < n; j++) {
                ++cnt[s[j] - 'a'];
                int mx = *max_element(begin(cnt), end(cnt));
                int mn = mx;
                for (const int& x : cnt)
                    if (x > 0)
                        mn = min(mn, x);
                res += mx - mn;
            }
        }
        return res;
    }
};

File: 1781. Sum of Beauty of All Substrings/Solution.py
class Solution:
    def beautySum(self, s: str) -> int:
        n, res = len(s), 0
        for i in range(n):
            cnt = [0] * 26
            for j in range(i, n):
                cnt[ord(s[j]) - ord("a")] += 1
                res += max(cnt) - min(e for e in cnt if e > 0)
        return res
